📖 release: restructure release docs team roles

dhij opened the pull request:
<!-- Thanks for sending a pull request! Here are some tips for you:
    1. If this is your first time, please read our contributor guidelines: https://github.com/kubernetes-sigs/cluster-api/blob/main/CONTRIBUTING.md#contributing-a-patch and developer guide https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/book/src/developer/guide.md

    2. Please add an icon to the title of this PR (see https://sigs.k8s.io/cluster-api/CONTRIBUTING.md#contributing-a-patch), and delete this line and similar ones
    the icon will be either ⚠️ (:warning:, major or breaking changes), ✨ (:sparkles:, feature additions), 🐛 (:bug:, patch and bugfixes), 📖 (:book:, documentation or proposals), or 🌱 (:seedling:, minor or other) 
-->

**What this PR does / why we need it**:

This PR breaks up the lengthy [release tasks doc](https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/release/release-tasks.md) by team as suggested in https://github.com/kubernetes-sigs/cluster-api/issues/10354

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #10354 

<!-- 
Please label this pull request according to what area(s) you are addressing. For reference on PR/issue labels, see: https://github.com/kubernetes-sigs/cluster-api/labels?q=area+

Area example:
/area runtime-sdk
-->

dhij commented:
/area documentation

chrischdi commented:
/ok-to-test

chrischdi commented on docs/release/role-handbooks/ci-signal/README.md:
```diff
@@ -0,0 +1,95 @@
+- [Tasks](#tasks)
+  - [Setup jobs and dashboards for a new release branch](#setup-jobs-and-dashboards-for-a-new-release-branch)
+  - [[Continuously] Monitor CI signal](#continuously-monitor-ci-signal)
+    - [[Continuously] Reduce the amount of flaky tests](#continuously-reduce-the-amount-of-flaky-tests)
```
```suggestion
  - [[Continuously] Reduce the amount of flaky tests](#continuously-reduce-the-amount-of-flaky-tests)
```

chrischdi commented:
> @chrischdi I fixed the broken links and would appreciate if you could have a look 🙏
> 
> When this PR gets merged, I will need a follow-up PR to fix a bunch of broken links to `release-tasks.md` since that file would no longer exist (ex. in `.github/ISSUE_TEMPLATE`). I saw that including those link changes in this PR will break the CI since those are absolute `https://` links.

Makes sense, maybe you can already prepare that PR so we can merge it quickly after this one to not break main for too long.

chrischdi commented on docs/release/release-tasks.md:
```diff
@@ -1,547 +0,0 @@
-   1. Host a release session over a public zoom meeting.
-   2. Record the session for future reference and transparency.
-   3. Use release process-related waiting periods as a forum for discussing issues/questions.
-   4. Publish the recording on YouTube channel.
-
-#### [Optional] [Track] Bump the Cluster API apiVersion
-
-**Note** This should only be done when we have to bump the apiVersion of our APIs.
-
-1. Add new version of the types:
-    1. Create new api packages by copying existing packages.
-    2. Make sure webhooks only exist in the latest apiVersion (same for other subpackages like `index`).
-    3. Add conversion and conversion tests.
-    4. Adjust generate targets in the Makefile.
-    5. Consider dropping fields deprecated in the previous apiVersion.
-2. Update import aliases in `.golangci.yml`.
-3. Switch other code over to the new version (imports across the code base, e.g. controllers).
-    1. Add all versions to the schema in the `main.go` files.
-4. Add types to the `PROJECT` files of the respective provider.
-5. Add test data for the new version in `test/e2e/data/{infrastructure-docker,shared}` (also update top-level `.gitignore`).
-6. Update `docker.yaml`, make sure all tests are successful in CI.
-
-#### [Optional] [Track] Bump the Kubernetes version
-
-1. Create an issue for the new Kubernetes version via: [New Issue: Kubernetes bump](https://github.com/kubernetes-sigs/cluster-api/issues/new/choose).
-2. Track the issue to ensure the work is completed in time.
-
-#### [Optional] Track Release and Improvement tasks
-
-1. Create an issue for easier tracking of all the tasks for the release cycle in question.
-   <br>Prior art: [Tasks for v1.6 release cycle](https://github.com/kubernetes-sigs/cluster-api/issues/9094)
-2. Create a release improvement tasks [GitHub Project Board](https://github.com/orgs/kubernetes-sigs/projects/55) to track 
-   the current status of all improvement tasks planned for the release, their priorities, status (i.e `Done`/`In Progress`)
-   and to distribute the work among the Release Team members.
-
-   **Notes**:
-   * At the beginning of the cycle, Release Team Lead should prepare the improvement tasks board for the ongoing release cycle.
-     The following steps can be taken:
-      - Edit improvement tasks board name for current cycle (e.g. `CAPI vX.Y release improvement tasks`)
-      - Add/move all individual missing issues to the board
```
I think the replacement for all of these are missing?

dhij replied:
@chrischdi Thank you for pointing that out, I missed those points. I have added them. I will add a summary of changes below 👍

sbueringer commented:
@dhij do you have time to address the finding?

dhij commented:
@sbueringer thank you for the reminder, i will have a look at it now 👀

dhij commented:
@chrischdi @sbueringer 👋 I have added the missing information and fixed the existing relative paths to the `release-tasks.md` file that does not exist anymore.

I also have a follow-up PR https://github.com/kubernetes-sigs/cluster-api/pull/10858 as Chris suggested to fix the other existing paths to `release-tasks.md`. Those changes could not be included in this same PR since they are absolute paths that will not exist until this PR is merged. Once this PR is merged, I will do a rebase on that PR 👍

chrischdi commented on docs/release/role-handbooks/release-lead/README.md:
```diff
@@ -0,0 +1,254 @@
+- [Tasks](#tasks)
+  - [Finalize release schedule and team](#finalize-release-schedule-and-team)
+  - [Add/remove release team members](#addremove-release-team-members)
+  - [Prepare main branch for development of the new release](#prepare-main-branch-for-development-of-the-new-release)
```
Doesn't this TOC miss items which are there? e.g.
`Optional] Track Release and Improvement tasks`

dhij replied:
Thank you for pointing that out, I addressed this in my most recent commits.

chrischdi commented on docs/release/release-team.md:
```diff
@@ -67,11 +67,17 @@ As noted above, making changes to  the CAPI release cadence is out of scope for
+
+* The examples in these documents are based on the v1.6 release cycle.
+* The documents focus on tasks that are done for every release. One-time improvement tasks are out of scope.
+* If a task is prefixed with `[Track]` it means it should be ensured that this task is done, but the folks with the corresponding role are not responsible to do it themselves.
```
Should we have this note in each handbook?

dhij replied:
I moved this bullet point to each of the handbooks in my most recent commits 👍

chrischdi commented on docs/release/release-team.md:
```diff
@@ -67,11 +67,17 @@ As noted above, making changes to  the CAPI release cadence is out of scope for
+**Notes**:
+
+* The examples in these documents are based on the v1.6 release cycle.
+* The documents focus on tasks that are done for every release. One-time improvement tasks are out of scope.
```
Does this still make sense as is?

dhij replied:
I removed this line as it is not relevant anymore.

sbueringer commented:
@dhij Do you have time to address the findings above?

dhij commented:
@sbueringer 👋 sorry for the delay, yes i will get to it 🙂

dhij commented:
@chrischdi @sbueringer 👋 i addressed your comments and would appreciate another review 🙏 

There were [3 new commits to the release-task.md](https://github.com/kubernetes-sigs/cluster-api/commits/main/docs/release/release-tasks.md) in the meantime and since the `release-task.md` file is removed as part of this PR, I had to manually cherry-pick the changes.

chrischdi commented:
/lgtm
/approve

Will do a small PR on-top after merging.
